third_party/move/tools/move-linter/src/model_ast_lints/nonminimal_bool.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module implements an expression linter that checks for boolean expressions:
+//! 1. `x && true`, which can be replaced with just `x`.
+//! 2. `x && false`, which can be replaced with just `false`.
+//! 3. `x || true`, which can be replaced with just `true`.
+//! 4. `x || false`, which can be replaced with just `x`.
+//! 5. `x <==> true`, which can be replaced with just `x`.
+//! 6. `x <==> false`, which can be replaced with just `!x`.
+//! 7. `x ==> true`, which can be replaced with just `true`.
+//! 8. `x ==> false`, which can be replaced with just `!x`.
+//! 9. `true ==> x`, which can be replaced with just `x`.
+//! 10. `false ==> x`, which can be replaced with just `true`.
+//! 11. `!true`, which can be replaced with just `false`.
+//! 12. `!false`, which can be replaced with just `true`.
+//!
+//! Note also that rules 1 through 6 have both LHS and RHS version
+
+use move_compiler_v2::external_checks::ExpChecker;
+use move_model::{
+    ast::{ExpData, Operation, Value},
+    model::GlobalEnv,
+};
+
+#[derive(Default)]
+pub struct NonMinimalBool;
+
+impl NonMinimalBool {
+    // Converts a binary boolean operator to its string representation
+    fn name_of_op(&self, cmp: &Operation) -> &'static str {
+        match cmp {
+            Operation::And => "&&",
+            Operation::Or => "||",
+            Operation::Implies => "==>",
+            Operation::Iff => "<==>",
+            _ => unreachable!("Unexpected operation"),
+        }
+    }
+
+    // Returns the message for a binary boolean operator with a literal
+    // cmp is the boolean operator
+    // b is the boolean literal, true or false
+    // side is 0 if literal is on the left and 1 if it is on the right
```

Comment:
nit: this could instead be a boolean param (say `literal_is_lhs` or similar) because it should only take two values, as opposed to `usize`.

Reply:
Done. I was considering this (as well as making it an enum).